### ISSUE TITLE
inconsistent kernel performance 

### ISSUE THREAD

**Issue by enjieli** [david-cortes/cmfrec#9]
@david-cortes, I am encountering some weird kernel behavior and hoping you can help me figure out why and how to solve this issue.

Quite often when I open my jupyter notebook and read in dataset, then run `CFM` OR `OFM` `model.fit` with the parameters that I previously had tested for successful model converge, I would get `ALS procedure failed` for CMF or failure message for OMF. But if I make a copy of the notebook and open in a new tab, then both models would converge with the same script and datasets. Also, this happens a lot when I did my parameter tuning for `OMF`, sometimes in a kernel, no parameter set would work, but in another kernel, the parameter set would work... Is this related with setting random seed?  I would really appreciate your help. Thank you so so much!

here is my code:
```python
import pickle
import pandas as pd
import numpy as np

#read in ratings data and item side info

ratings = pickle.load( open( "../CMFREC_dataset/ratings.pkl", "rb" ))

item_side_info = pickle.load( open( "../CMFREC_dataset/item_side_info.pkl", "rb" ))
```
```python
from sklearn.model_selection import train_test_split

#split item into two set, one for training and one for testing
items_train, items_test = train_test_split(ratings.ItemId.unique(), test_size=0.2, random_state=50)


#=================================
# ratings_test1:users and items which were both in the training data.

ratings_train, ratings_test1 = train_test_split(ratings.loc[ratings.ItemId.isin(items_train)],
                                                test_size=0.2, random_state=123)


item_attr_train = item_side_info.loc[item_side_info.ItemId.isin(items_train)]



#=================================
# ratings_test_2: users which were in the training data and items which were not in the training data.
ratings_test2 = ratings.loc[ratings.ItemId.isin(items_test)]


### Handy usage of Pandas indexing
item_attr_test = item_side_info.set_index("ItemId")
```

```python
from cmfrec import CMF_explicit

m_collective = CMF_explicit(k=100, w_main=0.5,w_item=0.5,niter=100,user_bias=False, item_bias=False)\
                .fit(X=ratings_train, I=item_attr_train)
```

Here is where the problems kick in, I often will get `ALS procedure failed` . But I know for a fact that those parameters had worked before.. and it would work if I open a new notebook...

**Comment by david-cortes** [comment]
Hmm that's weird, and it shouldn't fail like that. The random seed should not have anything to do with failing to fit the model. I guess it might have something to do with the inputs that you're passsing:
* Are you passing a COO matrix with some rows or columns having zero entries?
* Are you passing a sparse input (COO or DataFrame) with NAN or infinite values?

**Comment by enjieli** [comment]
I tried to set random seed at the beginning of my Jupyter notebook but that did not change anything. 

also, I noticed something weird going on in my terminal. It seems like if the Jupyter runs under yellow colored notebook, then it is doomed to fail, if it runs under green color kernel, then my model runs fine.  Any idea what's going on? ![Screen Shot 2020-04-29 at 4 06 06 PM](https://user-images.githubusercontent.com/5032864/80655425-7d65a500-8a33-11ea-90c7-5a270db80b4d.png)

**Comment by david-cortes** [comment]
No idea. Looks like some jupyter extension is failing, or that jupyter is failing to use it. Also you probably shouldn't use 100 ALS iterations.

**Comment by enjieli** [comment]
i am passing a data frame without any NAN, it seems like it is related with my jupyter notebook..
![Screen Shot 2020-04-29 at 4 06 06 PM](https://user-images.githubusercontent.com/5032864/80656257-95d6bf00-8a35-11ea-976b-a34e528c0c36.png)

**Comment by enjieli** [comment]
fixed the problem by reinstall ipykernel.
